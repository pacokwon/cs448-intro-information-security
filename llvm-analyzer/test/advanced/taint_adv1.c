#include "homework.h"

int main() {
	int x = source(); 
	int y = sanitizer(x);
	int z = y;
	int result;

	if(x > 0){
		x = y + z;
		result = x;
		sink(result);
	}
	else{
		result = x + y;
		sink(result);
	}

	return 0;
}
