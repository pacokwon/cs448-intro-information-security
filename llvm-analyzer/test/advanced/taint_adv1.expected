Potential Tainted-flow @ advanced/taint_adv1.c:main:16:3 (call void @sink(i32 %add2), !dbg !27)
